### ISSUE TITLE
MisconfigurationError: No TPU devices were found even when TPU is connected

### ISSUE THREAD

**Issue by adamDhalla** [PyTorchLightning/pytorch-lightning#5207]
Have been frustrated over the past few hours over a problem, though It's likely its a problem I started myself hah.

I'm trying to connect to the TPU in Colab. I'm pretty sure I've gotten all the import stuff down. My code is [here](https://colab.research.google.com/drive/1EaCJpyYK8xLuzO79ESAsIlGZAcWjOh8u?usp=sharing). I'm not completely set on everything, so the entire document isn't functional, but you should be able to see my attempts at TPU connection.

I'm running Pytorch in version 1.5.0 and torchvision in 0.6.0 because I was finding I couldn't install XLA with anything later than 1.5.0. I'm running XLA in version 20200325. 

This is the image that seems so confusing: It states that we have a connection with xla: 1 yet when trying to flag it in the trainer I get an error saying no TPUs can be found. 

![image](https://user-images.githubusercontent.com/69402866/102712990-c47e0000-4279-11eb-8ecd-d8dc98982dbb.png)

If anyone could help me, that would be amazing.
Thanks, 
A

**Comment by rohitgr7** [comment]
the notebook is a bit long to find the issue. Mind reproduce it with https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing??

**Comment by zcain117** [comment]
@rohitgr7 here is a repro using TheBoringModel: https://colab.sandbox.google.com/drive/1VPjiQj6VitN-losLO5J5IEuepcFa6Meu#scrollTo=AAtq1hwSmjKe

I added a cell at the beginning as described in https://pytorch-lightning.readthedocs.io/en/latest/tpu.html#colab-tpus

**Comment by rohitgr7** [comment]
@zcain117 not able to open it. Mind update the permissions??

**Comment by zcain117** [comment]
Here is a version using the cell from latest documentation: https://colab.research.google.com/drive/1tBExbwIy9NoP-YdWcnLBLRZtmsTIsV5g#scrollTo=DEi2_1B4Krf3

Here is a version using torch-xla-1.7: https://colab.research.google.com/drive/1BdRiZeX-S9RlFONyvlR-VexELrcAhYaX#scrollTo=DEi2_1B4Krf3

**Comment by rohitgr7** [comment]
both are working fine for me. I just enabled TPU env in colab and set `Trainer(..., tpu_cores=8)`.

**Comment by zcain117** [comment]
Thanks for checking. I changed nothing at all and now these are working for me too :)

What did you mean by `restart the notebook first after installing xla_wheel`? Is this the same as restarting the runtime?

I think I did some version of this too, the `1.7` version crashed a few times and I went through a few iterations of restarting the runtime and/or refreshing the colab page and now the notebook runs.

Maybe there is some intermittent problem TPU initialization behind the scenes in Colab. Sorry for the false alarm
